x-pack/legacy/plugins/infra/public/components/logging/log_text_stream/column_headers.tsx

```diff
@@ -35,7 +37,16 @@ export const LogColumnHeaders: React.FunctionComponent<{
               data-test-subj="logColumnHeader timestampLogColumnHeader"
               key={columnConfiguration.timestampColumn.id}
             >
-              Timestamp
+              <WithLogPosition>
+                {({ firstVisiblePosition }) => {
+                  if (firstVisiblePosition) {
+                    // return useFormattedTime(firstVisiblePosition.time, { show: 'date' });
```

Comment:
You'd use this commented-out line, and remove the one beneath, right?

Reply:
Yes! I want to, but the hooks only work in functional components, not in render callbacks. When I use the hook the app crashes.

I should have added a comment there for reviewers :)

Reply:
Oh, right. Perhaps you could extract that out to a component

```typescript
function FormattedTime({ time }: { time: number }) {
  const formatted = useFormattedTime(time, { show: 'date' });

  return formatted;
}
```

Then call it?
```tsx
<WithLogPosition>
  {({ firstVisiblePosition }) => {
    if (firstVisiblePosition) {
      return <FormattedTime time={firstVisiblePosition.time} />
    }

    return 'Timestamp';
  }}
</WithLogPosition>
```

Reply:
I've ended up changing the approach for the date after this commend from Jason https://github.com/elastic/kibana/issues/45986#issuecomment-537186091